### PR TITLE
Add a listAliases muxrpc endpoint

### PR DESCRIPTION
**Pull request by boreq**
I would like to get some feedback on adding a function similar to this one. This would make it possible to retrieve a list of all aliases for a particular user. This list could be used by users to preview their aliases without using the web interface.

### CODE REVIEW

**Comment by boreq** [comment]
I was considering making this a source request so that a lot of results could be returned efficiently. I don't know if this would make sense.

**Comment by staltz** [comment]
Hey @boreq, this PR seems fine, but it would need tests. I haven't worked directly on muxrpc APIs in go-ssb, but the code looks fine in my eyes. I'll still check if others Gophers have time to review this: @decentral1se @cblgh .

**Comment by boreq** [comment]
> Hey @boreq, this PR seems fine, but it would need tests.

Sure, I just wanted to get feedback if this would make any sense, the code was broken actually. I just added tests and fixed the code.

File: muxrpc/handlers/alias/handler.go [hunk 1]
```diff
@@ -147,3 +147,47 @@ func (h Handler) Revoke(ctx context.Context, req *muxrpc.Request) (interface{},
...
+		return nil, fmt.Errorf("listAlias: invalid feed ref: %w", err)
+	}
+
+	allAliases, err := h.db.List(ctx)
+	if err != nil {
+		var takenErr roomdb.ErrAliasTaken
+		if errors.As(err, &takenErr) {
+			return nil, takenErr
+		}
+		return nil, fmt.Errorf("listAlias: could not register alias: %w", err)
```
Comment (decentral1se) [inline muxrpc/handlers/alias/handler.go:174]: We're not trying to register, we're trying to list?

**Review by decentral1se — COMMENTED** [review]
LGTM! Nice one :rainbow:

File: muxrpc/handlers/alias/handler.go (hunk 1 above)
Comment (boreq) [reply muxrpc/handlers/alias/handler.go]: Right, this error handling in general was just bollocks, I removed it. Sorry, I whipped this out pretty quickly. There are some mistakes like this with error strings in other handlers as well which I will fix in a separate PR.

Comment (boreq) [reply muxrpc/handlers/alias/handler.go]: 6b932f2

Comment (decentral1se) [reply muxrpc/handlers/alias/handler.go]: Nice one, thanks!

**Comment by staltz** [comment]
Thanks for the PR and thanks for the review!

**Pull request merged**